### PR TITLE
Added Documentation for Depth of tree

### PR DESCRIPTION
**Pull request by Saatvik21**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

`Yes`

### Description

I've added an article for depth of tree.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes: #392

### CODE REVIEW

**Comment by Saatvik21** [comment]
@HarshCasper @iamrajiv @atarax665 @gat786 Please review!

File: docs/data-structures/depth-of-tree.mdx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  * Now just by looking at the definition, we can conclude that the maximum depth of the tree is 3 as there are 3 edges between the root node and the furthest leaves. 
+
+<p align="center">
+  <img width="700" height="400" src="https://i.imgur.com/xl6FpfC.png" />
+</p>
+
+- **Depth of a Node**
+  * According to the definition, we can find every node's depth by counting the number of edges from the root node to that node. In the above example, as we can see, `Node A` is the root node, so the number of edges in the path from the root node that is Node A to Node A is 0. So the depth of Node A is `0`.
+  * Similarly for `Node B`, the number of edges in the path from Node A to Node B is 1. So the depth of Node B is `1`. 
+  * We can find the depth of the rest of the nodes in the similar fashion.
```
Comment (iamrajiv) [inline docs/data-structures/depth-of-tree.mdx:29]:
```suggestion
  * We can find the depth of the rest of the nodes similarly.
```

File: docs/data-structures/depth-of-tree.mdx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+<p align="center">
+  <img width="700" height="400" src="https://i.imgur.com/jcB5zNa.png" />
+</p>
+
+## Algorithm
+
+In this section, we will discuss about the algorithmic steps performed to find the depth of tree.
```
Comment (iamrajiv) [inline docs/data-structures/depth-of-tree.mdx:37]:
```suggestion
In this section, we will discuss the algorithmic steps performed to find the depth of the tree.
```

File: docs/data-structures/depth-of-tree.mdx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+* If the root is null, then the depth of the tree is 0.
+* Else, start from the root and, 
+  * Recursively find the maximum depth of the left sub-tree.
+  * Recursively find the maximum depth of right sub-there.
+* Obtain the larger depth out of them and increment it by one.
+
+#### Depth of a Node
+
+* If the root is null, return -1.
+* If the root is equal to the value of the node who's depth we are supposed to find, print depth = 0.
```
Comment (iamrajiv) [inline docs/data-structures/depth-of-tree.mdx:50]:
```suggestion
* If the root is equal to the value of the node whose depth we are supposed to find, print depth = 0.
```

File: docs/data-structures/depth-of-tree.mdx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+        //if tree if not empty
+        else
+        {
+            node *temp =root;
+            while(1)
+            {
+                //if data to be inserted is less then the data of current node
+                if(x < temp->data )
+                {
+                    //if left child of current node is node present we simple add new node with the data to be inserted
```
Comment (iamrajiv) [inline docs/data-structures/depth-of-tree.mdx:104]:
```suggestion
                    // if a left child of the current node is node present we simply add a new node with the data to be inserte
```

File: docs/data-structures/depth-of-tree.mdx [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+                        {
+                            node *tnode = new node(x);
+                            temp->left_child = tnode;
+                            break;
+                        }
+                    //else we change the current node pointer with pointer of left child node pointer
+                    else
+                        temp = temp->left_child;
+                }
+                //if data to be inserted is grater then or equal to the data of current node
```
Comment (iamrajiv) [inline docs/data-structures/depth-of-tree.mdx:115]:
```suggestion
                // if data to be inserted is greater than or equal to the data of the current node
```

File: docs/data-structures/depth-of-tree.mdx [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+                else
+                {
+                    //if right child of current node is node present we simple add new node with the data to be inserted
```
Comment (iamrajiv) [inline docs/data-structures/depth-of-tree.mdx:118]:
```suggestion
                    // if a right child of the current node is node present we simply add a new node with the data to be inserted
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@Saatvik21 Implement the suggested changes.

**Comment by Saatvik21** [comment]
@HarshCasper @iamrajiv Done the required changes, Please review!

**Review by iamrajiv — APPROVED** [review]

File: docs/data-structures/depth-of-tree.mdx [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+```
+Depth of tree : 3
+
+         50                         level 1
+      /     \
+     /       \
+    40        60                    level 2
+   /   \     /   \
+  /     \   /     \
+30    45  55      70                level 3
```
Comment (HarshCasper) [inline docs/data-structures/depth-of-tree.mdx:246]: Is this even a part of the output?

File: docs/data-structures/depth-of-tree.mdx [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+
+tree = Tree(1)
+
+tree.root.left = TreeNode(2)
+tree.root.right = TreeNode(3)
+tree.root.left.left = TreeNode(4)
+tree.root.left.right = TreeNode(5)
+tree.root.right.left = TreeNode(6)
+tree.root.right.right = TreeNode(7)
+
```
Comment (HarshCasper) [inline docs/data-structures/depth-of-tree.mdx:214]: Don't use static input. Make it dynamic and ensure that the I/O sample for both C++ and Python matches.

File: docs/data-structures/depth-of-tree.mdx [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+  * Recursively find the maximum depth of the left sub-tree.
+  * Recursively find the maximum depth of right sub-there.
+* Obtain the larger depth out of them and increment it by one.
+
+#### Depth of a Node
+
+* If the root is null, return -1.
+* If the root is equal to the value of the node whose depth we are supposed to find, print depth = 0.
+* Else, recursively check if the node is present in either of the sub-trees.
+* If found, calculate the distance between the root node and the given node.
```
Comment (HarshCasper) [inline docs/data-structures/depth-of-tree.mdx:52]: Use backticks around numbers

File: docs/data-structures/depth-of-tree.mdx [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+  <img width="700" height="400" src="https://i.imgur.com/jcB5zNa.png" />
+</p>
+
+## Algorithm
+
+In this section, we will discuss the algorithmic steps performed to find the depth of the tree.
+
+#### Maximum Depth of Tree
+
+* If the root is null, then the depth of the tree is 0.
```
Comment (HarshCasper) [inline docs/data-structures/depth-of-tree.mdx:41]: Use backticks around numbers

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/data-structures/depth-of-tree.mdx (hunk 7 above)
Comment (Saatvik21) [reply docs/data-structures/depth-of-tree.mdx:246]: No, it's not. I just added it for the explanation, but I'll remove it now

**Review by iamrajiv — APPROVED** [review]

**Comment by Saatvik21** [comment]
@HarshCasper I made the python input to dynamic input, please check if it's right. Also my checks are failing now, so should I create a new PR or will this work too?

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**